### ISSUE TITLE
Cursor jumps to end of file when saving and "fix on save" is checked

### ISSUE THREAD

**Issue by zyklus** [AtomLinter/linter-jscs#98]
If this is an artifact of re-writing the whole file I'd suggest saving and restoring the cursor position.  It's especially annoying when the save doesn't actually change anything!

And on that note, can I suggest a basic comparison?

```js
var reformatted = jscs.format(contents);
if (contents !== reformatted){
    // replace file contents
}
```

This would also prevent needless entries in the undo history.

**Comment by acusti** [comment]
The behavior only occurs when the `fixString` method doesn’t change the file, so the comparison check you suggest actually avoids this issue. It’s fixed by c8bd3253d0d855802e6b0666808212a94fec08c5, which is a part of PR #87.

**Comment by jrust** [comment]
While it no longer moves the cursor if there are no changes, I'm still seeing a jump in 3.1.1 if it fixes something.

**Comment by acusti** [comment]
In my experience, the cursor jumps to the end of the file whenever `fixString` changes the current line that the cursor is on but retains its position otherwise. Worthwhile to fix, but I suppose it will require some logic to detect when the current line is changed and try to guess the best new cursor position.

**Comment by iam4x** [comment]
Closed with #125
